iroh-net/src/discovery/local_swarm_discovery.rs

```diff
@@ -213,38 +204,27 @@ impl LocalSwarmDiscovery {
                 .send_blocking(Message::Discovery(node_id.to_string(), peer.clone()))
                 .ok();
         };
-        let mut addrs: HashMap<u16, Vec<IpAddr>> = HashMap::default();
-        let mut has_ipv4 = false;
-        let mut has_ipv6 = false;
-        for socketaddr in socketaddrs {
-            if !has_ipv6 && socketaddr.is_ipv6() {
-                has_ipv6 = true;
-            };
-            if !has_ipv4 && socketaddr.is_ipv4() {
-                has_ipv4 = true;
-            };
-            addrs
-                .entry(socketaddr.port())
-                .and_modify(|a| a.push(socketaddr.ip()))
-                .or_insert(vec![socketaddr.ip()]);
-        }
-
-        let ip_class = match (has_ipv4, has_ipv6) {
-            (true, true) => IpClass::V4AndV6,
-            (true, false) => IpClass::V4Only,
-            (false, true) => IpClass::V6Only,
-            // this case indicates no ip addresses were supplied, in which case, default to ipv4
-            (false, false) => IpClass::V4Only,
-        };
+        let addrs = LocalSwarmDiscovery::socketaddrs_to_addrs(socketaddrs);
         let mut discoverer =
             Discoverer::new_interactive(N0_LOCAL_SWARM.to_string(), node_id.to_string())
                 .with_callback(callback)
-                .with_ip_class(ip_class);
+                .with_ip_class(IpClass::Auto);
```

Comment:
not sure I like this change too much, since what this does underneath is try to bind to a port which is later dropped. Sometimes this ends up with absurd race conditions about the port being taken. I'm going to go ahead with trusting this, but just leaving this here for the record I guess

Reply:
Not sure whether there may be a misunderstanding: the code tries binding to both ports and uses whatever worked — no successes are dropped. Is there a race condition in this that I’m unaware of?